tests/flytekit/unit/core/test_type_hints.py

```diff
@@ -1342,6 +1342,30 @@ def t2() -> dict:
     assert output_lm.literals["o0"].scalar.generic == expected_struct
 
 
+def test_guess_dict4():
```

Comment:
lets have an example of nested dataclass

```
@dataclass
class Nested():
   ...

@dataclass
class Outer():
    n: Nested
```

Reply:
sure